scarb/src/core/resolver.rs

```diff
@@ -38,4 +39,57 @@ impl Resolve {
             .iter(&self.graph)
             .unique()
     }
+
+    pub fn solution_with_target_kind(
+        &self,
+        root_package: PackageId,
+        target_kind: TargetKind,
+    ) -> Vec<PackageId> {
```

Comment:
Maybe we could optimise it to not collect vec? 🤔

Reply:
why not?

Reply:
The iterator references values owned by `filtered_graph` variable, I haven't wrapped my head how to reorganize the code yet

Reply:
In general, the solutions here won't be massive, and it's not really that important to try to optimize out this vec collection. If that's hard, then let's skip trying to do that now.